### ISSUE TITLE
Compiler should not report Warning RZ2012 when using Blazor bind syntax

### ISSUE THREAD

**Issue by bonsall** [dotnet/roslyn#76356]
**Version Used**: C# 12, C# 13 (potentially older C# versions as well)

**Steps to Reproduce**:

1. Create a blazor component that is setup to leverage the `@bind-VariableName` syntax that blazor provides, by adding `Parameter` properties that also have the `EditorRequiredAttribute` supplied. For example: `[Parameter, EditorRequired] public string Value { get; set; }`, `[Parameter, EditorRequired] public EventCallback<string> ValueChanged { get; set; }`, and `[Parameter, EditorRequired] public Expression<Func<string>> ValueExpression { get; set; }`
2. Use this blazor component anywhere in the application and use the bind syntax to set the variables `<MyComponent @bind-Value="SomeVariable" />`
3. Compiler will produce warning RZ2012 for both the `ValueChanged` and `ValueExpression` parameters, which is incorrect since those parameters are being set.

EX:
**SomeInput.razor**
```razor
@* Any Markup, it does't really matter *@
@code {
    [Parameter, EditorRequired]
    public string? Value { get; set; }

    [Parameter, EditorRequired]
    public EventCallback<string?> ValueChanged { get; set; }

    [Parameter, EditorRequired]
    public Expression<Func<string?>> ValueExpression { get; set; }
}
```

**App.razor**
```razor
<SomeInput @bind-Value="MyValue" />

@code {
    public string? MyValue { get; set; }
}
```
**Diagnostic Id**: RZ2012

**Expected Behavior**: No RZ2012 warning is produced.

**Actual Behavior**: An RZ2012 warning is produced.

**Comment by jjonescz** [comment]
This is fixed by https://github.com/dotnet/razor/pull/11043 which will be in a future .NET 9.0.200 release.
